ci/test/04_install.sh

```diff
@@ -11,6 +11,7 @@ if [[ $QEMU_USER_CMD == qemu-s390* ]]; then
 fi
 
 if [ "$CI_OS_NAME" == "macos" ]; then
+  sudo -H pip3 install --upgrade pip
```

Comment:
Is using `sudo -H` definitely needed here?

Reply:
Yes sudo -H is necessary, otherwise if I am  upgrading it using --user flag, it is giving some deprecated warning

Reply:
Also for multiprocess build, pip is directly being downloaded from ubuntu repositories, as will the case for most of the new ubuntu based builds. So, I think we should not upgrade its version in every CI build.